xlators/features/locks/src/posix.c

```diff
@@ -1351,14 +1358,22 @@ pl_getxattr(call_frame_t *frame, xlator_t *this, loc_t *loc, const char *name,
     int32_t op_errno = EINVAL;
     int32_t op_ret = -1;
     dict_t *dict = NULL;
+    char *client_uid = NULL;
+    pid_t client_pid;
 
     if (!name)
         goto usual;
 
     if (strncmp(name, GF_XATTR_CLRLK_CMD, SLEN(GF_XATTR_CLRLK_CMD)))
         goto usual;
 
-    op_ret = pl_getxattr_clrlk(this, name, loc->inode, &dict, &op_errno);
+    if (frame->root && frame->root->client) {
```

Comment:
If any of these are not present, we shouldn't allow any intrlk command. The function expects these values to be set.